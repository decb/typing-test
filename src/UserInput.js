import React from "react";

function UserInput(props) {
  const [buffer, setBuffer] = React.useState("");
  const [input, setInput] = React.useState("");

  const handleInput = event => {
    const newInput = event.target.value;

    if (newInput.length > 0 && newInput[newInput.length - 1] === " ") {
      setInput("");
      setBuffer(buffer + newInput);
    } else {
      setInput(newInput);
    }
  }

  props.setCount(buffer.split(/ +/).length - 1);
  props.setProgress(input);

  return (
    <input type="text" value={input} onChange={handleInput} />
  );
}

export default UserInput;
