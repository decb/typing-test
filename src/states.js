export const appStates = {
  NONE_SELECTED: 0,
  LOADING: 1,
  READY: 2,
  ERROR: 3
};

export const gameStates = {
  BEFORE: 0,
  DURING: 1,
  OVER: 2
};
