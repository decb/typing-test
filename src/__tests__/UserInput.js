import React from "react";
import renderer from "react-test-renderer";

import UserInput from "../UserInput";
import { gameStates } from "../states";

describe("<UserInput />", () => {
  test("Renders correctly", () => {
    const tree = renderer
      .create(
        <UserInput
          gameState={gameStates.BEFORE}
          startGame={jest.fn()}
          addInputWord={jest.fn()}
          setProgress={jest.fn()}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
