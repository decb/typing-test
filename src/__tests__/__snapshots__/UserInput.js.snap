// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserInput /> Renders correctly 1`] = `
<input
  autoCapitalize="off"
  autoComplete="off"
  autoCorrect="off"
  className="ant-input"
  disabled={false}
  onChange={[Function]}
  onKeyDown={[Function]}
  spellCheck="false"
  style={
    Object {
      "borderBottomRightRadius": 0,
      "borderTopRightRadius": 0,
    }
  }
  type="text"
  value=""
/>
`;
